packages/block-library/src/template-part/edit/index.js

```diff
@@ -247,10 +285,36 @@ export default function TemplatePartEdit( {
 									} )
 								}
 							>
-								{ __( 'Edit' ) }
+								{ hasNavigationBlocks
+									? sprintf(
+											/* translators: %s: template part title or fallback text */
+											__( 'Edit %s' ),
+											(
+												title || __( 'template part' )
+											).toLowerCase()
+									  )
+									: __( 'Edit' ) }
```

Comment:
I could see the `Edit %s` being helpful even if there isn't a navigation. It's certainly clearer. That way when you're looking to edit a template, it's always the same regardless if there's a navigation block inside the template or not. It's more consistent.